### PR TITLE
Group HTMLNode module

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

The AST in the namespace is not right according to the `psr-4`, there was no directory name with such a name. 

## 🔖 Changes
- Created a AST directory as "module wrapper" for them.
- Added `readonly` to HtmlNode properties.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
> The AST in the namespace is not right according to the psr-4, there was no directory name with such a name.

That's a mistake, it should be under "AST" directory.

HTMLNode is one of the nodes under AST. There will possibly other nodes and other logic related with AST.

**Comment by Chemaclass** [comment]
Then I will rename the dir from `HTMLNode` to `AST`.

File: src/AST/ValueObject/HtmlNode.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-namespace HtmlParser\ValueObject;
+namespace HtmlParser\AST\ValueObject;
```
Comment (Chemaclass) [inline src/AST/ValueObject/HtmlNode.php:5]: Should the `ValueObject` been under the `AST` directory/namespace or outside?

File: src/AST/HTMLNodeFactory.php [hunk 2]
```diff
@@ -4,19 +4,20 @@
...
 use Nette\Utils\FileSystem;
-use Nette\Utils\Strings;
+
+use function iterator_to_array;
 
 final class HTMLNodeFactory
 {
     /**
-     * @return HtmlNode[]
+     * @return list<HtmlNode>
```
Comment (TomasVotruba) [inline src/AST/HTMLNodeFactory.php:20]: We use convection syntax of type[]

File: src/AST/ValueObject/HtmlNode.php (hunk 1 above)
Comment (TomasVotruba) [reply src/AST/ValueObject/HtmlNode.php:5]: This looks good 👍

File: src/AST/HTMLNodeFactory.php (hunk 2 above)
Comment (Chemaclass) [reply src/AST/HTMLNodeFactory.php]: Done ✔️

**Comment by TomasVotruba** [comment]
Thank you :clap:

**Pull request merged**